### DISCUSSION TITLE
New user, odd problems

### DISCUSSION THREAD

**Discussion by kimjohn1** [SCIInstitute/SCIRun#2243] [General] [answered]
In the past 4 days I have installed SCIRun 5 on several computers operating with Ubuntu and Linux Mint (various versions).  All are working fine, so congratulations on whatever changes you have done recently.

I am working my way through the Basic Tutorial, and have discovered that attempting to save a network using "save as..." causes the application to become unresponsive.  About half the time the computer system becomes unresponsive, and requires a hard crash recovery (power button held in the off position until the system powers off).  Can you think of anything that might be causing this, or what I might do to investigate or work around the issue?

Also, I notice that execution takes 10 to 30 seconds to update the view after making a change in the network.  Is there a way to tell the application to use all available cores during operation, or is it already doing that?  Are there operator techniques I can use to speed up operation?  I am working on getting a more functional GPU for the PC that will be used for SCIRun investigation.  Do you have recommendation for what to get?

Kim

**Comment by dcwhite** [comment, marked as answer]
First question, is this on a native Linux install or one of these emulator setups you mentioned? That might severely impact disk speed for network saving. I have noticed some performance problems on Linux recently though, in some very basic places like latvol computation and ShowField. Those are new since I fixed the build. Your GPU weakness is probably contributing (how much video RAM do you have?), but since these issues I've seen impact a lot of basic areas I'm definitely going to investigate soon. As for the network saving problem, I have not tested that, and seems more likely due to the setup, but it could also be a new bug. I will take a look this week and see what I can replicate.

**Reply by kimjohn1** [reply]
No emulators.  All installations were on bare metal.

In an effort to get things working better I just installed an Nvidia GeForce 1660 Super (6 GB RAM, 1408 cores) on the computer that seemed to be having the most trouble.  Selecting "save as ..." no longer causes crashing, but it still has some issues; takes a long time to show the save file dialog, and does not show text when entering a file name to be saved.  I works, its just a little quirky.  I am more concerned about the performance.

Can you confirm for me that taking 30 to 60 seconds to update the view after making some change to the pipeline network is abnormal?  This is my first time using SCIRun so I don't know what would be normal performance.

I call up the nvidia-settings display and monitor GPU performance as I work with SCIRun, and I see no indication that the GPU is being tasked when I click on Execute All Modules, or update the view from within a module.  The gross delay is only present when I am working with a network that loads data from an external source.  If the network is using self generated data (CreateLatVol, as described in the Tutorial Chapter 3), then updating the view is instantaneous.  Since the "save as..." option and the ReadField module are both referencing local storage, could the long delays with both of those operations be manifestations of some common problem?

Kim

**Reply by dcwhite** [reply]
I have a fix for the dreadful performance of ShowField and some other modules that will be merged soon, check out https://github.com/SCIInstitute/SCIRun/pull/2244. 

For data being slow to and from disk (which would explain both of your issues), that's a bit harder to explain. No regressions to report there. Maybe also a hardware performance issue? You can print out a bit more logging information by running `./SCIRun --verbose`. If you're still having issues after #2244 is in master branch, I can add some extra logging for file operations for you to log what's going on.

**Reply by kimjohn1** [reply]
I am experiencing the ShowField wheel of eternal waiting as well.  Is a master with your Issue #2244 fix, available?

**Reply by kimjohn1** [reply]
I ran a simple network based on your Defibrillation Tutorial, and using the verbose flag.  The verbose output (all after the file is loaded) is provided as an attachment.
[SCIRun-verbose-see-events-at-17-02-11.txt](https://github.com/SCIInstitute/SCIRun/files/6167657/SCIRun-verbose-see-events-at-17-02-11.txt)
The events show a 29 second time gap at 17:02:11, which is the time spent waiting for the spinning wheel in the ShowField icon.  Have a look at the event sequence, and let me know if this is the issue you have fixed, or if there is something else I might need to investigate.

Kim

**Reply by kimjohn1** [reply]
Correction:  the network I loaded for the verbose capture was the Basic Tutorial, Chapter 2, engine.nhdr exercise.
-KJ

**Reply by dcwhite** [reply]
#2244 should be merged into master by the end of today. If you want to test it right this instant, you can `git checkout disable-stop-code`

The file load at the start of the file looks pretty fast. That ShowField execution ought to go back to normal with #2244 fix.

**Reply by kimjohn1** [reply]
All good!  My first opportunity to give it a whirl will be this evening.  I'll report what luck then.
Kim

**Reply by kimjohn1** [reply]
Success here with the SCIRun-master retrieved from github at 9:30 pm, 19 March.  Operation is smooth and quick.  I still have about a 10 second delay when I open ShowField Options and click on Show Edges, but I presume that is some specific issue not addressed by your latest fix.  Selections made with ShowField Options for Show Faces and Show Nodes are instantaneously rendered in the ViewScene display.  All other view affecting changes that I have tried have been instantly shown in ViewScene when they are applied.  Very nice.

I have experienced a bit of strangeness with the ShowField option for Enable Transparency with Edges and Faces.  The Enable Transparency option is selectable, and has an effect, but the amount of transparency is not affected by the enterable numeric value.  The feature appears to be using some constant value instead of the entered value.  The Enable Transparency function for Nodes does not suffer this behavior and appears to be fully functional.

So, this issue is resolved as far as I'm concerned, and I'm pressing on with using SCIRun 5 as the topic of a term project.

Kim

**Reply by dcwhite** [reply]
I think there's a bug for that--I'll try to find it in the list.
